source/val/validate.cpp

```diff
@@ -237,7 +244,64 @@ spv_result_t ValidateBinaryUsingContextAndValidationState(
   }
 
   for (size_t i = 0; i < vstate->ordered_instructions().size(); ++i) {
-    const auto& instruction = vstate->ordered_instructions()[i];
+    auto& instruction = vstate->ordered_instructions()[i];
+
+    {
```

Comment:
This is yucky, but changing ordered_instruction to provide a non-const instruction has ripples....